openwisp_controller/config/api/serializers.py

```diff
@@ -376,3 +377,34 @@ def update(self, instance, validated_data):
         instance = super().update(instance, validated_data)
         self._save_m2m_templates(instance)
         return instance
+
+
+class ReversionSerializer(BaseSerializer):
+    user_id = serializers.SerializerMethodField()
+    date_created = serializers.DateTimeField(
+        source="revision.date_created", read_only=True
+    )
+    comment = serializers.CharField(source="revision.comment", read_only=True)
+    content_type = serializers.SerializerMethodField()
+
+    class Meta:
+        model = Version
+        fields = [
+            "id",
+            "revision_id",
+            "object_id",
+            "content_type",
+            "db",
+            "format",
+            "serialized_data",
+            "object_repr",
+            "user_id",
+            "date_created",
```

Comment:
Thinking out loud: 

Do we really need to expose all these details in the API? The admin views for listing versions don't show all these details. I think we can hide internal details (e.g. db, format, object_repr, content_type)

Reply:
I think that if it's not expensive in terms of processing (number of queries, CPU usage), we can return this info in the API.
It doesn't have to be identical to the admin, it's ok to have more technical details in the REST API.
What would not be allowable in my opinion would be the opposite: having the REST API show less details than the admin.